### DISCUSSION TITLE
Array shows position ID instead of string value

### DISCUSSION THREAD

**Discussion by kruppy** [encode/django-rest-framework#8155] [Question & Answer] [answered]
Hi, 

I have 2 models, Job and Topic:

```python
class Job(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4(), editable=False)
    job_title = models.CharField(max_length=255)
    topic = models.ManyToManyField(to=Topic, blank=True)
 

class Topic(models.Model):
    name = models.CharField(max_length=255)

```

When trying to expose it as an API via DRF, I get following JSON:
```JSON
[
    {
        "id": "fdca2bd2-b3b0-453b-a1a3-a52ca061e665",
        "job_title": "Build MVP",
        "topic": [
            1, 
            2,

        ],
        
    }
]
```
But  I want "topic" to contain the String-values instead of "1", "2".

Do I have to do something in serializer? The jobs_serializer.py looks like that:

```python
class JobSerializer(serializers.ModelSerializer):

    class Meta:
        model = Job
        fields = '__all__'
```

Thanks in advance!

**Comment by lovelydinosaur** [comment, marked as answer]
The `SlugRelatedField` and `StringRelatedField` [examples in the docs](https://www.django-rest-framework.org/api-guide/relations/#stringrelatedfield) are both close to what you're looking for.

* `SlugRelatedField` can support a read-write field although you'd need name to be a unique model field if that's what you need, eg... `name = models.CharField(max_length=255, unique=True)`
* `StringRelatedField` is read-only, and would need you to define a `__str__` method on the `Topic` model.

**Reply by kruppy** [reply]
Thank you, it worked!
